pkg/storage/unified/resource/index.go

```diff
@@ -98,29 +107,37 @@ func (i *Index) Init(ctx context.Context) error {
 		}
 	}
 
+	end := time.Now().Unix()
+	if IndexServerMetrics != nil {
+		IndexServerMetrics.IndexCreationTime.WithLabelValues().Observe(float64(end - start))
+	}
+
 	return nil
 }
 
 func (i *Index) Index(ctx context.Context, data *Data) error {
-	res, err := getResource(data.Value.Value)
+	// Transform the raw resource into a more generic indexable resource
+	res, err := NewIndexedResource(data.Value.Value)
 	if err != nil {
 		return err
 	}
-	tenant := tenant(res)
-	i.log.Debug("indexing resource for tenant", "res", res, "tenant", tenant)
+	tenant := res.Namespace
+	i.log.Debug("indexing resource for tenant", "res", string(data.Value.Value), "tenant", tenant)
 	shard, err := i.getShard(tenant)
 	if err != nil {
 		return err
 	}
-	// Transform the raw resource into a more generic indexable resource
-	indexableResource, err := NewIndexedResource(data.Value.Value)
+	err = shard.index.Index(res.Uid, res)
 	if err != nil {
 		return err
 	}
-	err = shard.index.Index(res.Metadata.Uid, indexableResource)
-	if err != nil {
-		return err
+
+	// record latency from when event was created to when it was indexed
+	latencySeconds := float64(time.Now().UnixMicro()-data.Value.ResourceVersion) / 1e6
```

Comment:
resourceversion is a timestamp?

Reply:
yeah, it is now bc of [this PR](https://github.com/grafana/grafana/pull/92638)